src/gui/newwizard/states/basiccredentialssetupwizardstate.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) Fabian Müller <fmueller@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "basiccredentialssetupwizardstate.h"
+#include "jobs/checkbasicauthjobfactory.h"
+#include "networkjobs/fetchuserinfojobfactory.h"
+
+namespace OCC::Wizard {
+
+BasicCredentialsSetupWizardState::BasicCredentialsSetupWizardState(SetupWizardContext *context)
+    : AbstractSetupWizardState(context)
+{
+    _page = new BasicCredentialsSetupWizardPage(_context->accountBuilder().serverUrl());
+}
+
+void BasicCredentialsSetupWizardState::evaluatePage()
+{
+    auto *basicCredentialsSetupWizardPage = qobject_cast<BasicCredentialsSetupWizardPage *>(_page);
+    OC_ASSERT(basicCredentialsSetupWizardPage != nullptr);
+
+    const QString username = basicCredentialsSetupWizardPage->username();
+    const QString password = basicCredentialsSetupWizardPage->password();
+
+    _context->accountBuilder().setAuthenticationStrategy(new HttpBasicAuthenticationStrategy(username, password));
+
+    if (!_context->accountBuilder().hasValidCredentials()) {
+        Q_EMIT evaluationFailed(tr("Invalid credentials"));
+    }
+
+    auto strategy = dynamic_cast<HttpBasicAuthenticationStrategy *>(_context->accountBuilder().authenticationStrategy());
```

Comment:
```suggestion
    auto strategy = qobject_cast<HttpBasicAuthenticationStrategy *>(_context->accountBuilder().authenticationStrategy());
```

Reply:
Strategy classes do not inherit `Q_OBJECT` (that would just introduce overhead without any benefit), thus `qobject_cast` cannot be used.